### DISCUSSION TITLE
[Android] MapView black after Intent.ACTION_OPEN_DOCUMENT

### DISCUSSION THREAD

**Discussion by moiatto** [mapsforge/vtm#1021] [Q&A] [answered]
I have more fragments.
when I open a picker with Intent.ACTION_OPEN_DOCUMENT in other fragment, then the MapView show black.

See this
[video](https://youtube.com/shorts/sk4jSjoi-dw?feature=share)

What can it be?

Declared (no matter if static or not)
```
private MapView mapView;
private MapPreferences mapPrefs;
private Map map;
```

```
@Override
public void onResume() {
  super.onResume();

  mapPrefs.load(map);
  mapView.onResume();
}

@Override
public void onPause() {
    super.onPause();
   
    mapPrefs.save(map);
    mapView.onPause();
}
```

**Comment by devemux86** [comment]
See the [GettingStarted](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/GettingStarted.java) example for the life cycle of the `MapView`.
It contains file picker.

There is also the [FragmentActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/FragmentActivity.java) example with fragments.

**Comment by moiatto** [comment]
Unfortunately none of this.

I have an activity that show 2 fragment:
- one is for mapview (mapFragment) always visible on the screen
- other fragment where a call the picker or another app (**the black map also happens if I call an app**)

I call the picker in this second fragment and not direct from activity.

I think the GL engine crashes or no longer finds the context.

**Comment by devemux86** [comment]
If destroy the fragment / MapView then need to create a new MapView,
as they work in the samples of [vtm-android-example](https://github.com/mapsforge/vtm/tree/master/vtm-android-example).

**Comment by moiatto** [comment]
I logged the life cycles:
after call the picker:
the Main Activity goes in pause->stop
the MapFragment with MapView goes in pause->stop.

**Comment by moiatto** [comment, marked as answer]
Solved badly with a trick.

from Activity:
Replaced MapFragment...onRestart()
Removed MapFragment...onPause()

but from the symptoms it seems openGL engine problem.

The issue occurs on external calls (Intent.ACTION_OPEN_DOCUMENT or launch app) from a fragment
